fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java

```diff
@@ -214,14 +214,16 @@ private PProxyResult sendProxyRequest(PProxyRequest request) throws UserExceptio
                     PProxyResult result;
                     try {
                         result = future.get(Config.routine_load_kafka_timeout_second, TimeUnit.SECONDS);
-                    } catch (TimeoutException e) {
+                    } catch (Exception e) {
                         LOG.warn("failed to send proxy request to " + address + " err " + e.getMessage());
-                        // When getting kafka info timed out, we tried again three times.
-                        if (++retryTimes > 3 || (retryTimes + 1) * Config.routine_load_kafka_timeout_second >
-                                                                        Config.routine_load_task_timeout_second) {
-                            throw e;
+                        if (e instanceof TimeoutException) {
+                            // When getting kafka info timed out, we tried again three times.
+                            if (++retryTimes > 3 || (retryTimes + 1) * Config.routine_load_kafka_timeout_second >
+                                                                            Config.routine_load_task_timeout_second) {
+                                throw e;
+                            }
+                            continue;
```

Comment:
not continue when catch other exception?

Reply:
Why not directly put thoes lines in `catch(TimeoutException)` block?